### PR TITLE
[Fixes #9188] Unable to clone/copy geoapps

### LINKED ISSUE #9188: Unable to clone/copy geoapps
## Expected Behavior

Users should be able to clone/copy geoapps

## Actual Behavior
An error is raised on copying geoapps
```
'NoneType' object has no attribute 'get_real_instance'
Traceback (most recent call last):
  File "/opt/geonode/geonode/resource/manager.py", line 477, in copy
    _resource = self._concrete_resource_manager.copy(instance, uuid=_resource.uuid, defaults=to_update)
  File "/opt/geonode/geonode/geoserver/manager.py", line 160, in copy
    if isinstance(_resource.get_real_instance(), Dataset):
AttributeError: 'NoneType' object has no attribute 'get_real_instance'
```

## Steps to Reproduce the Problem

  1. Filter resources by dashboards
  2. Select and view a dashboard, Click on Save > Save As
  3. Notice that a new dashboard is not created.

## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):
  - Platform:
  - Additional details:

### PR DESCRIPTION
**Pull request by afabiani**
References: #9188

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/resource/manager.py [hunk 1]
```diff
@@ -447,23 +447,21 @@ def copy(self, instance: ResourceBase, /, uuid: str = None, owner: settings.AUTH
...
+                    _resource.save()
                     if isinstance(instance.get_real_instance(), Document):
-                        _resource.save()
                         for resource_link in DocumentResourceLink.objects.filter(document=instance.get_real_instance()):
                             _resource_link = copy.copy(resource_link)
                             _resource_link.pk = _resource_link.id = None
                             _resource_link.document = _resource.get_real_instance()
                             _resource_link.save()
                     if isinstance(instance.get_real_instance(), Dataset):
                         _resource.name = f'{_resource.name}_{uuid1().hex[:8]}'
```
Comment (marthamareal) [inline geonode/resource/manager.py:458]: @afabiani the  name updated here is not going to be saved

**Review by mattiagiupponi — APPROVED** [review]

File: geonode/resource/manager.py [hunk 2]
```diff
@@ -447,23 +447,21 @@ def copy(self, instance: ResourceBase, /, uuid: str = None, owner: settings.AUTH
                     _resource = copy.copy(instance.get_real_instance())
                     _resource.pk = _resource.id = None
                     _resource.uuid = uuid or str(uuid4())
+                    _resource.save()
```
Comment (marthamareal) [inline geonode/resource/manager.py:450]: We also get an error on saving the dataset `DETAIL:  Key (store, workspace, name)=(geonode_data, geonode, {name}) already exists.`

**Review by marthamareal — CHANGES_REQUESTED** [review]

**Comment by marthamareal** [comment]
Also for maps, clones don't have the given title because `AttributeError: 'Map' object has no attribute 'name'`
raised [here](https://github.com/GeoNode/geonode/blob/master/geonode/resource/manager.py#L502)

File: geonode/resource/manager.py (hunk 1 above)
Comment (afabiani) [reply geonode/resource/manager.py]: Thanks @marthamareal pushed a fix and test case for this one

File: geonode/resource/manager.py (hunk 2 above)
Comment (afabiani) [reply geonode/resource/manager.py:452]: Thanks @marthamareal pushed a fix and test case for this one

**Comment by afabiani** [comment]
> Also for maps, clones don't have the given title because `AttributeError: 'Map' object has no attribute 'name'` raised [here](https://github.com/GeoNode/geonode/blob/master/geonode/resource/manager.py#L502)

Done thanks

File: geonode/resource/manager.py [hunk 3]
```diff
@@ -447,23 +447,22 @@ def copy(self, instance: ResourceBase, /, uuid: str = None, owner: settings.AUTH
                     _resource = copy.copy(instance.get_real_instance())
                     _resource.pk = _resource.id = None
                     _resource.uuid = uuid or str(uuid4())
+                    # Avoid integrity errors...
+                    _resource.name = f'{_resource.name}_{uuid1().hex[:8]}'
```
Comment (marthamareal) [inline geonode/resource/manager.py:451]: Map throws `AtributeError: 'Map' object has no attribute 'name'` here

**Review by marthamareal — APPROVED** [review]

**Review by mattiagiupponi — APPROVED** [review]

**Review by giohappy — APPROVED** [review]

**Pull request merged**